templates/parachain/runtime/src/genesis_config_presets.rs

```diff
@@ -0,0 +1,163 @@
+
+use crate::{AccountId, SessionKeys, Signature, EXISTENTIAL_DEPOSIT};
+use alloc::{format, vec, vec::Vec};
+use serde_json::Value;
+use sp_core::{sr25519, Pair, Public};
+use sp_runtime::traits::{IdentifyAccount, Verify};
+
+/// Preset configuration name for a local testnet environment.
+pub const PRESET_LOCAL_TESTNET: &str = "local_testnet";
+
+/// Preset configuration name for a development environment.
+pub const PRESET_DEVELOPMENT: &str = "development";
+
+type AccountPublic = <Signature as Verify>::Signer;
+
+/// The default XCM version to set in genesis config.
+const SAFE_XCM_VERSION: u32 = xcm::prelude::XCM_VERSION;
+
+/// Helper function to generate a crypto pair from seed
+pub fn get_from_seed<TPublic: Public>(seed: &str) -> <TPublic::Pair as Pair>::Public {
+	TPublic::Pair::from_string(&format!("//{}", seed), None)
+		.expect("static values are valid; qed")
+		.public()
+}
+
+/// Generate collator keys from seed.
+///
+/// This function's return type must always match the session keys of the chain in tuple format.
+pub fn get_collator_keys_from_seed(seed: &str) -> AuraId {
+	get_from_seed::<AuraId>(seed)
+}
+
+/// Helper function to generate an account ID from seed
+pub fn get_account_id_from_seed<TPublic: Public>(seed: &str) -> AccountId
+where
+	AccountPublic: From<<TPublic::Pair as Pair>::Public>,
+{
+	AccountPublic::from(get_from_seed::<TPublic>(seed)).into_account()
+}
+
+/// Generate the session keys from individual elements.
+///
+/// The input must be a tuple of individual keys (a single arg for now since we have just one key).
+pub fn template_session_keys(keys: AuraId) -> SessionKeys {
+	SessionKeys { aura: keys }
+}
+
+fn testnet_genesis(
+	invulnerables: Vec<(AccountId, AuraId)>,
+	endowed_accounts: Vec<AccountId>,
+	root: AccountId,
+	id: ParaId,
+) -> Value {
+	serde_json::json!({
+		"balances": {
+			"balances": endowed_accounts.iter().cloned().map(|k| (k, 1u64 << 60)).collect::<Vec<_>>(),
+		},
+		"parachainInfo": {
+			"parachainId": id,
+		},
+		"collatorSelection": {
+			"invulnerables": invulnerables.iter().cloned().map(|(acc, _)| acc).collect::<Vec<_>>(),
+			"candidacyBond": EXISTENTIAL_DEPOSIT * 16,
+		},
+		"session": {
+			"keys": invulnerables
+				.into_iter()
+				.map(|(acc, aura)| {
+					(
+						acc.clone(),                 // account id
+						acc,                         // validator id
+						template_session_keys(aura), // session keys
+					)
+				})
+			.collect::<Vec<_>>(),
+		},
+		"polkadotXcm": {
+			"safeXcmVersion": Some(SAFE_XCM_VERSION),
+		},
+		"sudo": { "key": Some(root) }
+	})
+}
+
+fn local_testnet_genesis() -> Value {
+	testnet_genesis(
+		// initial collators.
+		vec![
+			(
+				get_account_id_from_seed::<sr25519::Public>("Alice"),
+				get_collator_keys_from_seed("Alice"),
+			),
+			(
+				get_account_id_from_seed::<sr25519::Public>("Bob"),
+				get_collator_keys_from_seed("Bob"),
+			),
+		],
+		vec![
+			get_account_id_from_seed::<sr25519::Public>("Alice"),
```

Comment:
Ah i have seen this crap in a lot of places being copy pasted 🙈 

Similarly, I have what I think is a better way in #5117:

1. `sp_keyring::Keyring::iter()`
2. re-export it from `frame` as well for better discover-ability.

Reply:
I like it. Not sure however, if `Keyring` actually shall go to frame?
Let's do this in the followup - there will following work for solochain and minimal.

Some common parts need to be extracted, probably the accounts in presets can be unified somehow.
(also see https://github.com/paritytech/polkadot-sdk/pull/4739#issuecomment-2157341035 which needs to be addressed).

Reply:
@iulianbarbu this is a pretty easy one for you to pick up :)

Reply:
@kianenigma I am a bit late to the party. Based on the context on this thread I get that we want to do something like `sp_keyring::Keyring::Alice.to_account_id()` for the specific line where the thread started, and in general, for creating vec's we want to create them based on `sp_keyring::Keyring::iter()`.

I do not get the context around `re-export it from 'frame' as well for better discover-ability`. Maybe we can sync offline (to not pollute anymore this merged PR) and maybe open an issue (if there isn't one) which clarifies next steps.

Reply:
_re-export it from 'frame' as well for better discover-ability_ basically means that you don't need extra `use` / crate dep when defining presets (`sp_keyring`), it  is just imported by `frame::runtime`.

Reply:
I see now. Thanks!

Reply:
Please also see https://github.com/paritytech/polkadot-sdk/pull/5327/files#r1758961989

Reply:
No, I am not doing a similar thing.

Reply:
hm, isn't this done here:
https://github.com/paritytech/polkadot-sdk/pull/6094/files#diff-b35b8d703bb99b1beb9205e7083c88898399cbee2107d730f5cb6984502e9e4cR409